src/test/java/com/example/moviemanager/controller/MovieControllerTest.java

```diff
@@ -0,0 +1,157 @@
+        String jsonMovie = objectMapper.writeValueAsString(movie);
+        when(movieService.save(movie)).thenReturn(movie);
+
+        // then
+        mockMvc.perform(post("/movies")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonMovie))
+                .andExpect(status().isCreated())
+                .andExpect(content().string(jsonMovie));
+    }
+
+    @Test
+    void deletes_movie_by_given_id() throws Exception {
+        // given
+        int movieId = 1;
+        doNothing().when(movieService).deleteById(movieId);
+
+        // then
+        mockMvc.perform(delete("/movies/{movieId}", movieId))
+                .andExpect(status().isOk())
+                .andExpect(content().string(blankString()));
+    }
+
+    @Test
+    void throws_exception_when_trying_to_delete_non_existing_movie() throws Exception {
+        // given
+        int movieId = 1;
+        String message = "Deletion failed. Could not find movie with id - " + movieId;
+
+        doThrow(new MovieNotFoundException(message)).when(movieService).deleteById(movieId);
+
+        // then
+        mockMvc.perform(delete("/movies/{movieId}", movieId))
+                .andExpect(status().isNotFound())
+                .andExpect(content().string(containsString(message)));
+    }
+
+    @Test
+    void updates_movie() throws Exception {
+        // given
+        int movieId = 1;
+        Movie movie = new Movie(5, "Home Alone", "Christmas movie", 1990, 8.5);
+        Movie expectedMovie = new Movie(movieId, "Home Alone", "Christmas movie", 1990, 8.5);
+        String jsonMovie = objectMapper.writeValueAsString(movie);
+        String expectedJsonMovie = objectMapper.writeValueAsString(expectedMovie);
+        when(movieService.update(movieId, movie)).thenReturn(expectedMovie);
+
+        // then
+        mockMvc.perform(put("/movies/{movieId}", movieId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonMovie))
+                .andExpect(status().isOk())
+                .andExpect(content().string(expectedJsonMovie));
+    }
+
+    @Test
+    void throws_exception_when_trying_to_update_non_existing_movie() throws Exception {
+        // given
+        int movieId = 1;
+        Movie movie = new Movie(5, "Home Alone", "Christmas movie", 1990, 8.5);
+        String jsonMovie = objectMapper.writeValueAsString(movie);
+        String message = "Update failed. Could not find movie by id - " + movieId;
+        doThrow(new MovieNotFoundException(message)).when(movieService).update(movieId, movie);
+
+        // then
+        mockMvc.perform(put("/movies/{movieId}", movieId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonMovie))
+                .andExpect(status().isNotFound())
+                .andExpect(content().string(containsString(message)));
+    }
+}
```

Comment:
Please add one more test when some unexpected exception is being thrown because you have such error handling too

Reply:
Included test with incorrect usage of endpoint.